src/test/java/uk/nhs/tis/trainee/notifications/service/PlacementServiceTest.java

```diff
@@ -464,4 +492,228 @@ void shouldHandleUnknownNotificationTypesForNotificationDays() {
+  @CsvSource(delimiter = '|', textBlock = """
+      email@example.com | PROTOCOL_EMAIL
+      https://example.com | ABSOLUTE_URL
+      not a href | NON_HREF""")
+  void shouldIncludeContactDetailsInInAppNotification(String contact, HrefType contactType)
+      throws SchedulerException {
+    Placement placement = new Placement();
+    placement.setTisId(TIS_ID);
+    placement.setPersonId(PERSON_ID);
+    placement.setStartDate(START_DATE);
+    placement.setOwner(OWNER);
+    placement.setPlacementType(IN_POST);
+    placement.setSpecialty(SPECIALTY);
+    placement.setSite(SITE);
+
+    when(notificationService.meetsCriteria(placement, true)).thenReturn(true);
+
+    List<Map<String, String>> contactList = List.of(
+        Map.of(CONTACT_TYPE_FIELD, LocalOfficeContactType.TSS_SUPPORT.getContactTypeName()));
+    when(notificationService.getOwnerContactList(OWNER)).thenReturn(contactList);
+    when(notificationService.getOwnerContact(contactList, LocalOfficeContactType.TSS_SUPPORT,
+        null)).thenReturn(contact);
+    when(notificationService.getHrefTypeForContact(contact)).thenReturn(
+        contactType.getHrefTypeName());
+
+    service.addNotifications(placement);
+
+    ArgumentCaptor<Map<String, Object>> variablesCaptor = ArgumentCaptor.forClass(Map.class);
+    verify(inAppService).createNotifications(eq(PERSON_ID), any(), eq(PLACEMENT_INFORMATION),
+        eq(PLACEMENT_INFO_VERSION), variablesCaptor.capture(), anyBoolean(), any());
+
+    Map<String, Object> variables = variablesCaptor.getValue();
+    assertThat("Unexpected variable count.", variables.size(), is(5));
+    assertThat("Unexpected specialty.", variables.get(PLACEMENT_SPECIALTY_FIELD),
+        is(SPECIALTY));
+    assertThat("Unexpected site.", variables.get(PLACEMENT_SITE_FIELD),
+        is(SITE));
+    assertThat("Unexpected start date.", variables.get(START_DATE_FIELD), is(START_DATE));
+    assertThat("Unexpected local office contact.", variables.get(LOCAL_OFFICE_CONTACT_FIELD),
+        is(contact));
+    assertThat("Unexpected local office contact type.",
+        variables.get(LOCAL_OFFICE_CONTACT_TYPE_FIELD), is(contactType.getHrefTypeName()));
+  }
+
+  @Test
+  void shouldNotAddInAppNotificationsWhenNotMeetsCriteria() throws SchedulerException {
+    Placement placement = new Placement();
+    placement.setTisId(TIS_ID);
+    placement.setPersonId(PERSON_ID);
+    placement.setStartDate(START_DATE);
+    placement.setOwner(OWNER);
+    placement.setPlacementType(IN_POST);
+    placement.setSpecialty(SPECIALTY);
+    placement.setSite(SITE);
+
+    when(notificationService.meetsCriteria(placement, true)).thenReturn(false);
+
+    service.addNotifications(placement);
+
+    verifyNoInteractions(inAppService);
+  }
+
+  @ParameterizedTest
+  @EnumSource(value = NotificationType.class, mode = EnumSource.Mode.INCLUDE,
+      names = {"PLACEMENT_INFORMATION"})
+  void shouldNotAddInAppNotificationsWhenNotUnique(NotificationType notificationType)
```

Comment:
Afaik, the only reason not to add an in-app notification is if one exists that has already been shown to the trainee (sentAt is in the past) - if that's correct, could you check the test below and maybe rename it to make it clearer what its checking?